akka-actor/src/main/scala/akka/actor/TypedActor.scala

```diff
@@ -152,11 +151,11 @@ object TypedActor extends ExtensionId[TypedActorExtension] with ExtensionIdProvi
       case ps if ps.length == 0 ⇒ SerializedMethodCall(method.getDeclaringClass, method.getName, method.getParameterTypes, Array())
       case ps ⇒
         val serialization = SerializationExtension(akka.serialization.JavaSerializer.currentSystem.value)
-        val serializedParameters = new Array[(Int, Class[_], Array[Byte])](ps.length)
+        val serializedParameters = new Array[(Int, String, Array[Byte])](ps.length)
         for (i ← 0 until ps.length) {
           val p = ps(i)
           val s = serialization.findSerializerFor(p)
-          val m = if (s.includeManifest) p.getClass else null
+          val m = Serializer.manifestFor(s, p).getOrElse("")
```

Comment:
Here is the main fix. There were no check for SerializerWithStringManifest and therefore wrong manifest was sent.